src/utils/userLocation.ts

```diff
@@ -1,81 +1,125 @@
-import firebase, { UserInfo } from "firebase/app";
-
-import { Room } from "types/rooms";
+import firebase from "firebase/app";
 
 import { updateUserProfile } from "pages/Account/helpers";
 import { useInterval } from "hooks/useInterval";
 
+import { LOCATION_INCREMENT_MS, LOCATION_INCREMENT_SECONDS } from "settings";
+
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+import { openRoomUrl } from "./url";
 
-const LOCATION_INCREMENT_SECONDS = 10;
-const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
+export type LocationData = Record<string, number>;
 
-type LocationData = Record<string, number>;
+export type UpdateLocationDataProps = {
+  userId: string;
+  newLocationData: LocationData;
+};
 
-export const updateLocationData = (
-  userId: string,
-  newLocationData: LocationData = {}
-) => {
+export const updateLocationData = ({
+  userId,
+  newLocationData,
+}: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
     lastSeenIn: newLocationData,
   });
 };
 
-export interface TrackLocationProps {
+export type SetLocationDataProps = {
   userId: string;
   locationName: string;
-  lastSeenIn?: LocationData;
-}
+};
 
-export const trackLocationEntered = ({
+export const setLocationData = ({
   userId,
   locationName,
-}: TrackLocationProps) => {
-  updateLocationData(userId, {
-    [locationName]: getCurrentTimeInMilliseconds(),
+}: SetLocationDataProps) => {
+  updateLocationData({
+    userId,
+    newLocationData: {
+      [locationName]: getCurrentTimeInMilliseconds(),
+    },
+  });
+};
+
+export type UpdateCurrentLocationDataProps = {
+  userId: string;
+  profileLocationData: LocationData;
+};
+
+export const updateCurrentLocationData = ({
+  userId,
+  profileLocationData,
+}: UpdateCurrentLocationDataProps) => {
+  const [locationName] = Object.keys(profileLocationData);
+
+  updateLocationData({
+    userId,
+    newLocationData: { [locationName]: getCurrentTimeInMilliseconds() },
   });
 };
 
-export const enterVenue = (venueId: string) => openUrl(venueInsideUrl(venueId));
+export const clearLocationData = (userId: string) => {
+  updateLocationData({ userId, newLocationData: {} });
+};
 
-type EnterExternalRoomProps = {
+export type EnterExternalRoomProps = {
   userId: string;
-  room: Room;
+  roomUrl: string;
+  roomSlug: string;
 };
 
-export const enterExternalRoom = ({ userId, room }: EnterExternalRoomProps) => {
-  trackLocationEntered({
+export const enterExternalRoom = ({
+  userId,
+  roomSlug,
+  roomUrl,
+}: EnterExternalRoomProps) => {
+  setLocationData({
     userId,
-    locationName: room.url,
+    locationName: roomSlug,
   });
```

Comment:
Since `roomSlug` is just passed into `locationName` (which seems to be our standard way of referring to 'location slugs' in the code here), and also represents the 'location slug', it would probably be useful to change this to `locationName` as well.

Reply:
fixed in 200c94f53638642cd3f70418f66b2accb8f392fe